services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/taydennys/TaydennysAttachmentService.kt

```diff
@@ -97,4 +97,18 @@ class TaydennysAttachmentService(
         }
         return newAttachment
     }
+
+    fun deleteAllAttachments(taydennys: TaydennysIdentifier) {
+        logger.info { "Deleting all attachments from täydennys. ${taydennys.logString()}" }
+        try {
+            metadataService
+                .deleteAllAttachments(taydennys)
+                .forEach(attachmentContentService::delete)
+            logger.info { "Deleted all attachments from täydennys. ${taydennys.logString()}" }
+        } catch (e: Exception) {
+            logger.error(e) {
+                "Failed to delete all attachment content for täydennys. Continuing with täydennyys deletion regardless of error. ${taydennys.logString()}"
+            }
+        }
```

Comment:
We don't know if the exception is from the metadata service or the content service. We shouldn't ignore errors from the metadata service, so that call should be before the try-catch:
```suggestion
        val paths = metadataService.deleteAllAttachments(taydennys)
        try {            
            paths.forEach(attachmentContentService::delete)
        } catch (e: Exception) {
            logger.error(e) {
                "Failed to delete all attachment content for täydennys. Continuing with täydennyys deletion regardless of error. ${taydennys.logString()}"
            }            
        }
        logger.info { "Deleted all attachments from täydennys. ${taydennys.logString()}" } 
```

Reply:
Right you are again. Thank you. Corrected it as suggested.